api/test/wfl/integration/slack_test.clj

```diff
@@ -0,0 +1,34 @@
+(ns wfl.integration.slack-test
+  (:require [clojure.test :refer :all]
+            [wfl.service.slack :as slack]
+            [wfl.util :as util])
+  (:import [clojure.lang PersistentQueue]))
+
+(def ^:private testing-agent (agent (PersistentQueue/EMPTY)))
+(def ^:private testing-slack-channel "C026PTM4XPA")
+(defn ^:private testing-slack-notification []
+  {:channel testing-slack-channel :message (format "WFL Integration Test Message: %s" (util/utc-now))})
+(def ^:private notify-promise (promise))
+(defn ^:private mock-send-notification
+  [queue]
+  (if (seq queue)
+    (let [{:keys [channel message]} (peek queue)
+          callback #(if (get % "ok")
+                      (deliver notify-promise true)
+                      (deliver notify-promise false))]
+      (slack/post-message channel message callback)
+      (pop queue))
+    queue))
+
+(add-watch testing-agent :watcher
+           (fn [_key _ref _old-state new-state]
+             (when (seq new-state)
+               (testing "notification is added to agent"
+                 (is (= (:channel (first (seq new-state))) testing-slack-channel))))))
```

Comment:
Why `(seq new-state)`?

Reply:
since the agent is created by `(agent (PersistentQueue/EMPTY))`, without `seq` it will not print something readable 🤔

Reply:
OK.